parsl/data_provider/files.py

```diff
@@ -46,9 +46,13 @@ def __init__(self, url, dman=None, cache=False, caching_dir="."):
         self.scheme = parsed_url.scheme if parsed_url.scheme else 'file'
         self.netloc = parsed_url.netloc
         self.path = parsed_url.path
+        self.registered = False
+        self.shared = False
         self.filename = os.path.basename(self.path)
         self.dman = dman if dman else DataManager.get_data_manager()
         self.data_future = {}
+        self.task_path = None
+        self.in_task = False
```

Comment:
@AndrewLitteken:

There is a mechanism using `local_path` to allow a new worker-side path to be specified by a staging mechanism. I've made PR #808 which makes this usable with all file schemes.

It might be enough that you can use that instead of `task_path` and `in_task`: once you set `local_path`, then asking for `file.filepath` will return that overridden path, which is what you're doing here. I'm a little unsure if you need to separately track `in_task` here. The other staging mechanisms don't do so.